File: Codes/2706-buy-two-chocolates.cpp
/*
 * @Author: LetMeFly
 * @Date: 2023-12-29 16:59:01
 * @LastEditors: LetMeFly
 * @LastEditTime: 2023-12-29 17:02:57
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        int m1 = 1000, m2 = 1000;  // 令m1 <= m2
        for (int p : prices) {
            if (p < m2) {
                m2 = p;
                if (m1 > m2) {
                    swap(m1, m2);
                }
            }
        }
        return m1 + m2 > money ? money : money - m1 - m2;
    }
};

File: Codes/2706-buy-two-chocolates.py
'''
Author: LetMeFly
Date: 2023-12-29 17:03:27
LastEditors: LetMeFly
LastEditTime: 2023-12-29 17:04:42
'''
from typing import List

class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        m1, m2 = 1000, 1000
        for p in prices:
            if p < m2:
                m2 = p
                if m1 > m2:
                    m1, m2 = m2, m1
        return money if m1 + m2 > money else money - m1 - m2
